test/widgets/lightbox_test.dart

```diff
@@ -152,5 +267,148 @@ void main() {
       await tester.tap(find.byIcon(Icons.play_circle_rounded));
       check(FakeVideoPlayerPlatform.isPlaying).isTrue();
     });
+
+    testWidgets('unsupported video shows an error dialog', (tester) async {
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestUnsupportedVideoUrl));
+      final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+      await tester.tap(find.byWidget(checkErrorDialog(tester,
+        expectedTitle: zulipLocalizations.errorDialogTitle,
+        expectedMessage: zulipLocalizations.errorVideoPlayerFailed)));
+    });
+
+    testWidgets('video advances overtime & stops playing when it ends', (tester) async {
+      await setupPage(tester, videoSrc: Uri.parse(FakeVideoPlayerPlatform.kTestVideoUrl));
+      check(FakeVideoPlayerPlatform.isPlaying).isTrue();
+      check(FakeVideoPlayerPlatform.position).equals(Duration.zero);
+
+      final halfTime = FakeVideoPlayerPlatform.kTestVideoDuration * 0.5;
+
+      await tester.pump(halfTime);
+      FakeVideoPlayerPlatform.pumpEvents();
+      checkPosition(tester, halfTime);
```

Comment:
In a real app, these events are sent from platform bindings, for example [here's android](https://github.com/flutter/packages/blob/260102b64c0fac9c66b7574035421fa6c09f5f89/packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java#L189-L196).

The initial revisions of this PR used `Timer.periodic` instead of `Stopwatch`, and in that implementation these events were dispatched from the `Timer.periodic` callback, but with `Stopwatch` there's no such hook thus the need for an explicit `pumpEvents` function.

(In this revision; Removed the usage from tests which don't rely on it.)